kubectl-plugin/pipeline/gc.go

```diff
@@ -88,6 +102,35 @@ func (o *gcOption) cleanPipelineRunInNamespace(namespace string) (err error) {
 	return
 }
 
+func ascOrderWithCompletionTime(items []unstructured.Unstructured) {
+	sort.Slice(items, func(i, j int) bool {
+		left := items[i]
+		right := items[j]
+
+		var leftCompletionTime time.Time
+		var rightCompletionTime time.Time
+		var err error
+
+		if leftCompletionTimeStr, ok, nestErr := unstructured.NestedString(left.Object, "status", "completionTime"); ok && nestErr == nil {
```

Comment:
I'd prefer to create a function to parse the completion time.

https://github.com/kubesphere-sigs/ks/blob/6c8251b66eefe48baa7235e0283b8e55b09b582e/kubectl-plugin/pipeline/gc.go#L114-L128

https://github.com/kubesphere-sigs/ks/blob/6c8251b66eefe48baa7235e0283b8e55b09b582e/kubectl-plugin/pipeline/gc.go#L135-L140

Reply:
Good catch!